kernel/src/scheduler/round_robin.rs

```diff
@@ -59,8 +59,12 @@ impl<'a> RoundRobinSched<'a> {
     /// How long a process can run before being pre-empted
     const DEFAULT_TIMESLICE_US: u32 = 10000;
     pub const fn new() -> RoundRobinSched<'a> {
+        Self::new_with_time(Self::DEFAULT_TIMESLICE_US)
+    }
+
+    pub const fn new_with_time(time_us: u32) -> RoundRobinSched<'a> {
         RoundRobinSched {
-            time_remaining: Cell::new(Self::DEFAULT_TIMESLICE_US),
+            time_remaining: Cell::new(time_us),
```

Comment:
This only influences the timeslice of the first execution unit scheduled. All subsequent schedules of this or other processes will have their timeslice reset to `Self::DEFAULT_TIMESLICE_US` in line 109.

I agree we want to make this configurable, so we should probably introduce another field in the `RoundRobinSched` struct for this. This will add some runtime cost the compiler is unlikely to optimize out (would need to change the struct layout), but that seems worth the tradeoff here.